### PR TITLE
Refactored the App class so it inherits from XCUIApplication now

### PR DESCRIPTION
**Pull request by zacoid55**
- Enhancement to do: Add the ability to pass launch arguments in
Changed BaseFeature to TestBase to better reflect its use
Changed the protocal in BasePage to enforce a trait on each page that inherits from it

### CODE REVIEW

File: Example/TABTestKit.xcodeproj/xcshareddata/xcschemes/TABTestKit_ExampleUITests.xcscheme [hunk 1]
```diff
@@ -26,6 +26,7 @@
       buildConfiguration = "Debug"
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
+      language = ""
```
Comment (KITSTABSamBerry) [inline Example/TABTestKit.xcodeproj/xcshareddata/xcschemes/TABTestKit_ExampleUITests.xcscheme]: what is this?

File: TABTestKit/Classes/BasePage.swift [hunk 2]
```diff
@@ -10,34 +10,24 @@
...
+  public init() { }
   
-  public override init() {}
-
   /// Native back button control
   public func tapBackButton() {
-    let app = XCUIApplication()
-    
-    app.navigationBars.children(matching: .button).matching(identifier: "Back").element(boundBy: 0).tap()
+    App.shared.navigationBars.children(matching: .button).matching(identifier: "Back").element(boundBy: 0).tap()
```
Comment (KITSTABSamBerry) [inline TABTestKit/Classes/BasePage.swift]: We probably don't want that in the pod as not every app will have this button.  Could we get rid of base page all together as this is the only function in it?

File: Example/TABTestKit.xcodeproj/xcshareddata/xcschemes/TABTestKit_ExampleUITests.xcscheme (hunk 1 above)
Comment (zacoid55) [reply Example/TABTestKit.xcodeproj/xcshareddata/xcschemes/TABTestKit_ExampleUITests.xcscheme]: I think this gets added with Swift 4 conversion? It's weird but it needs to be there for some reason

File: TABTestKit/Classes/BasePage.swift (hunk 2 above)
Comment (zacoid55) [reply TABTestKit/Classes/BasePage.swift]: Surely most apps have this? Isn't it a required Apple thing?

Comment (zacoid55) [reply TABTestKit/Classes/BasePage.swift]: @KITSTABSamBerry do we run the risk of this library/pod not needing to exist if we strip everything out?

**Pull request merged**